google/cloud/bigquery/enums.py

```diff
@@ -480,3 +480,18 @@ class SourceColumnMatch(str, enum.Enum):
     NAME = "NAME"
     """Matches by name. This reads the header row as column names and reorders
     columns to match the field names in the schema."""
+
+
+class TimestampPrecision(object):
```

Comment:
Looking at the code again after our talk, I still think this should be turned into a true enum.

With the current code, TimestampPrecision is just a shortcut to creating int or None values. [The argument type can't be TimeStampPrecision](https://github.com/googleapis/python-bigquery/pull/2333/files#r2604634140). This means all the docstrings have to describe all the values as `Union[int | None]`, and then do more work to describe how to build one or what those special values mean, and we lose a lot of the value of encapsulating it in an enum to begin with

If we change this to actually be an enum, we will have to convert it back to it's raw value before sending it to the backend. But it should make typing a lot cleaner

Reply:
But if you want to leave it as-is, that's fine. We'd just need to update the typing to be accurate